velox/exec/AggregationHook.h

```diff
@@ -99,6 +101,9 @@ class AggregationHook : public ValueHook {
 
 namespace {
 template <typename TValue, bool Overflow>
+#if defined(FOLLY_DISABLE_UNDEFINED_BEHAVIOR_SANITIZER)
+FOLLY_DISABLE_UNDEFINED_BEHAVIOR_SANITIZER("signed-integer-overflow")
```

Comment:
This looks like a common file, would this affect other functions as well apart from the spark_sum() function?

Reply:
No, it shouldn't affect other functions. `FOLLY_DISABLE_UNDEFINED_BEHAVIOR_SANITIZER("signed-integer-overflow")` applies only to this method and this method is only used in SumHook that is used for pushdown of SumAggregateBase. This hook (as well as SumAggregateBase) is used by both Presto and Spark sum() functions. But here the signed integer overflow may happen at line 111 that is only executed on integers when Overflow is set to true for Spark. So this change doesn't affect Presto's sum() function.

Reply:
> Also, is it possible to conditionally disable this sanitizer when Overflow template variable is true?

Unfortunately, `FOLLY_DISABLE_UNDEFINED_BEHAVIOR_SANITIZER ` only applies to the entire C function. But I think turning off ubsan for the entire C function is fine. When Overflow is false (i.e., for Presto's sum()), this method takes the code path `result = checkedPlus<TValue>(result, value);`. checkedPlus doesn't cause signed integer overflow. Instead, it throws a VELOX_ARITHMETIC_ERROR. So disabling ubsan in this method should not affect the code path taken by Presto.